pkgs/development/node-packages/aliases.nix

```diff
@@ -0,0 +1,38 @@
+pkgs: lib: self: super:
+
+### Deprecated aliases - for backward compatibility
+
+with self;
+
+let
+  # Removing recurseForDerivation prevents derivations of aliased attribute
+  # set to appear while listing all the packages available.
+  removeRecurseForDerivations = alias: with lib;
+    if alias.recurseForDerivations or false
+    then removeAttrs alias ["recurseForDerivations"]
+    else alias;
+
+  # Disabling distribution prevents top-level aliases for non-recursed package
+  # sets from building on Hydra.
+  removeDistribute = alias: with lib;
+    if isDerivation alias then
+      dontDistribute alias
+    else alias;
+
+  # Make sure that we are not shadowing something from node-packages.nix.
+  checkInPkgs = n: alias:
+    if builtins.hasAttr n super
+    then throw "Alias ${n} is still in node-packages.nix"
+    else alias;
```

Comment:
I've just realised a slight issue with this. Because `node-packages.nix` is generated by running the `generate.sh` script, I can't realistically remove the package from there without re-running the generation script. That's quite inconvenient to do. In my opinion it would make more sense check that the package is not present in `node-packages.json` since it will then just get deleted next time the codegen script is ran, or just not check it at all.

Reply:
That makes sense. I'd originally figured people could just remove the individual entry in node-packages.nix by hand, but it is a large unwieldy file and alternative to manual editing, letting that script run for hours is hell

Maybe I could make a quick script to remove singular attrs from node-packages.nix?

Otherwise yeah I guess I'll do the json check or remove the check

Reply:
the check should be kept otherwise aliases might be added which don't have any effect.

Reply:
> the check should be kept otherwise aliases might be added which don't have any effect.

What do you mean by "not have any effect"? Cause atm the check is not stopping anyone from aliasing a package that has never existed. All they do is prevent you from leaving some of the dead code in `node-packages.nix` around

Reply:
Is a short script to remove entries from both `node-packages.nix` and `node-packages.json` sufficient for you @ehllie?

I made a very hacky one last night and can open a PR with the script and instructions to use it in the comment in `aliases.nix`

Reply:
I've actually realised it's pretty simple to remove the package from `node-packages.nix`. I just never bothered to actually inspect the contents of the files `node2nix` generates all that much, and assumed it would be more complicated. But a script to do that automatically, or a quick explanation on how to do it would be nice nonetheless.

Reply:
I've opened #230782 with the hacky script